### PR TITLE
Fix undefined sanitizer issue in compact_array

### PR DESCRIPTION
**Pull request by MBkkt**
Also I think really correct code, should start with non zero capacity (so correct capacity can be seted in ctor)

### CODE REVIEW

File: src/s2/util/gtl/compact_array.h [hunk 1]
```diff
@@ -413,7 +413,9 @@ class compact_array_base {
     value_allocator_type allocator;
 
     T* new_ptr = allocator.allocate(capacity());
-    memcpy(new_ptr, Array(), old_capacity * sizeof(T));
+    if (old_capacity != 0) {
+      memcpy(new_ptr, Array(), old_capacity * sizeof(T));
+    }
     allocator.deallocate(Array(), old_capacity);
```
Comment (MBkkt) [inline src/s2/util/gtl/compact_array.h:419]: Maybe better? I don't read class code

```suggestion
    if (old_capacity != 0) {
      memcpy(new_ptr, Array(), old_capacity * sizeof(T));
      allocator.deallocate(Array(), old_capacity);
    }
```

**Comment by jmr** [comment]
> Also I think really correct code, should start with non zero capacity (so correct capacity can be seted in ctor)

Why? Feel free to send another PR if you think there are more improvements.

**Comment by MBkkt** [comment]
> Why? Feel free to send another PR if you think there are more improvements

Because we always have some capacity on "stack". Same as for sso string.

**Comment by jmr** [comment]
> Because we always have some capacity on "stack". Same as for sso string.

I'm not sure what you mean. There's already inline SSO-like space allocated: https://github.com/google/s2geometry/blob/e15d00b21e9bbc252e436c243710cb9712465814/src/s2/util/gtl/compact_array.h#L124

Send a PR to make it clear.

**Comment by MBkkt** [comment]
I mean sso string after default ctor have non zero capacity.
Why same not possible for this array?

**Comment by jmr** [comment]
Anything is possible. Send a PR if you think it can be improved.

**Pull request merged**